pvlib/irradiance.py

```diff
@@ -2819,3 +2820,130 @@ def dni(ghi, dhi, zenith, clearsky_dni=None, clearsky_tolerance=1.1,
             (zenith < zenith_threshold_for_zero_dni) &
             (dni > max_dni)] = max_dni
     return dni
+
+
+def pvfactors_bifacial_irradiance(
+        solar_azimuth, solar_zenith, surface_azimuth, surface_tilt,
+        timestamps, dni, dhi, gcr, pvrow_height, pvrow_width, albedo,
+        n_pvrows=3, index_observed_pvrow=1,
+        rho_front_pvrow=0.03, rho_back_pvrow=0.05,
+        horizon_band_angle=15.,
+        run_parallel_calculations=True, n_workers_for_parallel_calcs=None):
+    """
+    Calculate front and back surface plane-of-array irradiance on
+    a fixed tilt or single-axis tracker PV array configuration, and using
+    the open-source "pvfactors" package.
+    Please refer to pvfactors' online documentation for more details:
+    https://sunpower.github.io/pvfactors/
+
+    Parameters
+    ----------
+    solar_azimuth: numpy array of numeric
```

Comment:
The rest of pvlib generally uses pandas.Series or pandas.Dataframes.  If pvfactors must have np.array as input, can you do some type conversion in this function?

Reply:
Thanks @cwhanse , I forgot about that convention. So I created a decorating function that does this (with a test for it) in order to separate the modeling logic from the inputs/outputs conversion logic. Let me know if that approach is okay with you

Reply:
Does pvfactors fail with a Series or DataFrame?

In general, I like the decorator approach if we can make it robust enough to be used throughout pvlib. (Applying the decorator throughout pvlib should be done in a separate PR.)

Reply:
pvfactors has been tested with numpy arrays so far. But I can try to see if it would work with Series and DataFrames, with some luck duck-typing might work...

Understood; I did implement a test for the decorating function which checks multiple cases. But I can bring the type conversion into the pvfactors function if you prefer. Let me know